### PR TITLE
tests: migrate from Travis to GitHub-Actions

### PR DESCRIPTION
**Pull request by kiranparajuli589**
### Description
- migrate from `travis` to `gh-actions`

### Motivation
- https://github.com/sabre-io/xml/pull/206

### Related
- fork pr: https://github.com/kiranparajuli589/vobject/pull/1 (you can see the build status here)

### CODE REVIEW

**Comment by phil-davis** [comment]
@DeepDiver1975 I thought that I had access to repo settings, but it seems not. Maybe you can remove Travis as an expected CI result, and enable GitHub Actions etc?

**Comment by staabm** [comment]
I just removed prettyCI and travisCI from the whole sabre-io.org

**Comment by phil-davis** [comment]
@staabm is there a way to get the GitHub actions to run on this PR? Or do we review and "just merge it" and then see if it works?

**Comment by staabm** [comment]
> GitHub actions to run on this PR? Or do we review and "just merge it" and then see if it works?

AFAIK the github actions would run within the PR immediately when the PR is created by a person with write-access to the repo.

for other people it runs only after merge

**Comment by phil-davis** [comment]
> for other people it runs only after merge

Yesterday I saw a button to click on PR https://github.com/sabre-io/xml/pull/204 to give it permission to run the GitHub Actions CI. (That PR was from a fork) I clicked it, CI ran, passed, then we merged,

But I don't see that option here. But maybe that is because this is the very first PR that is actually implementing the GitHub actions.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Code Analysis (PHP CS-Fixer)
+        if: matrix.code-analysis == 'yes'
+        run: php vendor/bin/php-cs-fixer fix --dry-run --diff
+
+      - name: Code Analysis (PHPStan)
+        if: matrix.code-analysis == 'yes'
+        run: composer phpstan
+
+      - name: Test with PHPUnit
+        run: composer phpunit
```
Comment (phil-davis) [inline .github/workflows/ci.yml:59]: IMO `composer phpunit` does not run with coverage, so we might not get any coverage data.

**Pull request merged**